modules/machineset-aws-existing-placement-group.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-machinesets/creating-machineset-aws.adoc
+// * machine_management/control_plane_machine_management/cpmso-using.adoc
+
+ifeval::["{context}" == "cpmso-using"]
+:cpmso:
+endif::[]
+
+:_content-type: PROCEDURE
+[id="machineset-aws-existing-placement-group_{context}"]
+= Assigning machines to placement groups for Elastic Fabric Adapter instances by using machine sets
+
+You can configure a machine set to deploy machines on link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html[Elastic Fabric Adapter] (EFA) instances within an existing AWS placement group.
+
+EFA instances do not require placement groups, and you can use placement groups for purposes other than configuring an EFA. This example uses both to demonstrate a configuration that can improve network performance for machines within the specified placement group.
+
+.Prerequisites
+
+* You have created a placement group in the AWS console.
++
+[NOTE]
+====
+Ensure that the link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/placement-groups.html#limitations-placement-groups[rules and limitations] for the type of placement group that you create are compatible with your intended use case.
+ifdef::cpmso[]
+The control plane machine set spreads the control plane machines across multiple failure domains when possible. To use placement groups for the control plane, you must use a placement group type that can span multiple Availability Zones.
+endif::cpmso[]
+====
+
+.Procedure
+
+. In a text editor, open the YAML file for an existing machine set or create a new one.
+
+. Edit the following lines under the `providerSpec` field:
++
+[source,yaml]
+----
+ifndef::cpmso[]
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+endif::cpmso[]
+ifdef::cpmso[]
+apiVersion: machine.openshift.io/v1
+kind: ControlPlaneMachineSet
+endif::cpmso[]
+# ...
+spec:
+  template:
+    spec:
+      providerSpec:
+        value:
+          instanceType: <supported_instance_type> # <1>
+          networkInterfaceType: EFA # <2>
```

Comment:
I don't think this is necessary to even specify. At least I don't remember doing that.

Reply:
To clarify: Not necessary because you can use a placement group without an EFA, or not necessary _even if_ you are using an EFA?

Reply:
Sorry, should have made this more clear - `networkInterfaceType: EFA` should not be necessary when NOT using EFA. So for this example, as you mentioned in your last comment, that should be good.

Reply:
As we are taking this part of docs as an example to use the placementGroups, I am good with that.

Reply:
ok cool, thanks for clarifying :)